### PR TITLE
KFSPTS 12390 Stale Check Job

### PR DESCRIPTION
**Pull request by ajd299**

### CODE REVIEW

**Comment by ajd299** [comment]
retest this please

File: src/main/java/com/rsmart/kuali/kfs/cr/batch/service/impl/StaleCheckExtractServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+
+    protected void validateCheckReconciliation(StaleCheckBatchRow staleCheckRow, CheckReconciliation checkReconciliation, List<String> validationErrors) {
+        if (!StringUtils.equalsIgnoreCase(staleCheckRow.getBankCode(), checkReconciliation.getBankCode())) {
+            validationErrors.add("The Bank Code in the file (" + staleCheckRow.getBankCode() + ") row does not match the bank on the Check (" +
+                    checkReconciliation.getBankCode() + ").");
+        }
+
+        try {
+            Double checkTotalAmountParsed = NumberFormat.getNumberInstance(java.util.Locale.US).parse(staleCheckRow.getCheckTotalAmount()).doubleValue();
+            KualiDecimal checkTotalAmount = new KualiDecimal(checkTotalAmountParsed);
```
Comment (cah292) [inline src/main/java/com/rsmart/kuali/kfs/cr/batch/service/impl/StaleCheckExtractServiceImpl.java]: KualiDecimal has a constructor that accepts a String as input, so you don't need to parse the String value with NumberFormat explicitly.

**Comment by ajd299** [comment]
retest this please

**Comment by ajd299** [comment]
retest this please

**Comment by ajd299** [comment]
retest this please

**Comment by ajd299** [comment]
retest this please

**Comment by ajd299** [comment]
Ready for review again

File: src/main/java/com/rsmart/kuali/kfs/cr/dataaccess/CheckReconciliationDao.java [hunk 2]
```diff
@@ -73,4 +73,6 @@
      * @return
      */
     public List<PaymentGroup> getAllPaymentGroupForSearchCriteria(KualiInteger disbNbr, Collection<String> bankCodes);
+
+    public CheckReconciliation findByCheckNumber(String checkNumber);
```
Comment (cah292) [inline src/main/java/com/rsmart/kuali/kfs/cr/dataaccess/CheckReconciliationDao.java]: Is the check number enough of a constraint for uniqueness when searching for CR rows? If we don't expect the bank account number to change, then I think we will be okay.

Comment (ajd299) [reply src/main/java/com/rsmart/kuali/kfs/cr/dataaccess/CheckReconciliationDao.java]: I'm not sure, would you recommend adding the bank code?

Comment (cah292) [reply src/main/java/com/rsmart/kuali/kfs/cr/dataaccess/CheckReconciliationDao.java]: I think adding bank code will be enough, unless the CR rows are using several different bank account numbers for the same bank (which I do not think is the case).

**Comment by ajd299** [comment]
Ready for re-review

**Pull request merged**